lib/elixir/lib/gen_event.ex

```diff
@@ -655,11 +655,12 @@ defmodule GenEvent do
   Record.defrecordp :handler, [:module, :id, :state, :pid, :ref]
 
   @doc false
-  def init_it(starter, :self, name, mod, args, options) do
+  def init_it(starter, :self, name, mod, args, options)
```

Comment:
Because this is a callback, we should make sure those things are validated in the entry-point (start_link) rather than here.